File: 2026/aed-1/cormen/12.1/Node.py
from typing import Any

class Node:
    def __init__(self, key: int, data: Any) -> None:
        self.key: int = key
        self.data: Any = data
        self.left: Node | None = None
        self.right: Node | None = None
        self.p: Node | None = None

File: 2026/aed-1/cormen/12.1/Node.java
public class Node<T> {
    public int key;
    public T data;
    public Node<T> left = null;
    public Node<T> right = null;
    public Node<T> p = null;

    Node(int key, T data) {
        this.key = key;
        this.data = data;
    }
}
